### ISSUE TITLE
Array of shape

### ISSUE THREAD

**Issue by nbgo** [dwightjack/vue-types#4]
Is it possible to do smth like this?
```
VueTypes.arrayOf(VueTypes.shape({
    startOn: VueTypes.instanceOf(Date).isRequired
}))
```

**Comment by dwightjack** [comment]
Hi,

Looking at the code I guess it should work, but since I don't have any test for it I'll check it.

**Comment by nbgo** [comment]
Yes, it works. Problem was that array element had an additional property which was not described in shape. Can be validation made a bit less strict to allow such properties?

**Comment by dwightjack** [comment]
Hi,

I believe that React PropTypes will fail as well. In my opinion `shape` should describe the full _shape_ of an object. Maybe I could add a `partialShape` type. In you scenario as workaround I suggest to pass to `arrayOf`a custom Vue validation object:

```
VueTypes.arrayOf({
    type: Object
    validate(value) {
        return value.startOn instanceof Date;
    }
})
```

**Comment by nbgo** [comment]
`partialShape` would be nice.

**Comment by dwightjack** [comment]
Hi, this should be closed by v0.6.0 whoch implements a `.loose` flag on the `.shape` type.

**Comment by nikolaybobrovskiy** [comment]
Great! Thank you!
